### PR TITLE
feat(secret): secrets pagination

### PR DESCRIPTION
**Pull request by peveuve**
# Description

Manage pagination when retrieving list of secrets.

# How Has This Been Tested?

Unit test checks the skiptoken (=startIndex) parameter is taken in account when retrieving secrets from the vault.

- [x] get secrets without startIndex
- [x] get secrets with startIndex

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: lib/domain/Vault.js [hunk 1]
```diff
@@ -15,9 +15,13 @@ class Vault extends EventEmitter {
     return `${protocol}://${hostname}/secrets/${secretName}/${secretVersion}`
   }
 
-  getSecrets (quantity) {
-    const secretsFound = this.secrets.slice(0, quantity)
-    return cloneDeep(secretsFound)
+  getSecrets (quantity = 25, startIndex = 0) {
```
Comment (franck-mahieu) [inline lib/domain/Vault.js:18]: Quelle est l'utilité de recupére une quantité de secret ? on ne veux pas soit un secret particulier, soit tous les secrets ?

File: lib/routes/secretRouter.js [hunk 2]
```diff
@@ -21,14 +21,15 @@ function findLastVersion (secret) {
 
 secretRouter.get('/', (req, res) => {
   const maxResults = Number.parseInt(req.query.maxresults) || 25
+  const nextIndex = Number.parseInt(req.query.skiptoken) || 0
   const vault = VaultFileClient.getInstance().getVault()
-  const secrets = vault.getSecrets(maxResults)
+  const secrets = vault.getSecrets(maxResults, nextIndex)
```
Comment (franck-mahieu) [inline lib/routes/secretRouter.js:26]: ça correspond à quoi `nextIndex` ?

File: lib/domain/Vault.js (hunk 1 above)
Comment (peveuve) [reply lib/domain/Vault.js:18]: c'est dans la spec Microsoft

File: lib/routes/secretRouter.js (hunk 2 above)
Comment (peveuve) [reply lib/routes/secretRouter.js:26]: l'indice auquel on va chercher la prochaine page de secrets

**Pull request merged**